### PR TITLE
COD-888: Update Parse-SDK-Android

### PR DESCRIPTION
**Pull request by arblitroshani**
#### Corresponding Ticket / Related Links

- [COD-888](https://mitarbeiterapp.atlassian.net/browse/COD-888)

### What has been done?

- Update Parse-SDK-Android to version 1.20.0

### Why was this done?

- The previous version of our trigger parse fork used Parse-SDK 1.18, which internally still uses the [deprecated jobdispatcher](https://github.com/parse-community/Parse-SDK-Android/blob/3d439ffb52521c8f1f3893b40a7bbedf9adb343f/gcm/build.gradle#L36).

### How to test the changes?

See testing instructions in [this frontend PR](https://github.com/Staffbase/frontend/pull/4330).

### CODE REVIEW

File: module/manifest.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
+			"minimum_version": "1.10.0",
+			"module": "bolts"
+		}
+	},
+	"description": "Parse.com configuration to enable push notifications.\n\nThe parse module uses the Parse.com SDK to allow for native push notifications.",
+	"min_platform_version": "v2.8.6",
+	"namespace": "parse",
+	"platform_version": "v2.8.6",
+	"version": "3.5.8"
+}
```
Comment (rkusa) [inline module/manifest.json:15]: Could you update the PR to now change the indentation for the JSON file here? 'cause git churn and 'cause all other JSON files in this repo use 4 spaces, too

File: inspector/ios-inspector/dist/build_steps/sb_parse.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    { "do": { "add_ios_system_framework": "CoreGraphics.framework" } },
+    { "do": { "add_ios_system_framework": "libsqlite3.dylib" } },
+    { "do": { "add_ios_system_framework": "libz.1.1.3.dylib" } },
+    { "do": { "add_ios_system_framework": "MobileCoreServices.framework" } },
+    { "do": { "add_ios_system_framework": "QuartzCore.framework" } },
+    { "do": { "add_ios_system_framework": "Security.framework" } },
+    { "do": { "add_ios_system_framework": "Social.framework" } },
+    { "do": { "add_ios_system_framework": "StoreKit.framework" } },
+    { "do": { "add_ios_system_framework": "SystemConfiguration.framework" } }
+]
```
Comment (rkusa) [inline inspector/ios-inspector/dist/build_steps/sb_parse.json:14]: Why are those additions necessary?

File: module/manifest.json [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
+			"maximum_version": "1.10.0",
+			"minimum_version": "1.10.0",
+			"module": "bolts"
+		}
+	},
+	"description": "Parse.com configuration to enable push notifications.\n\nThe parse module uses the Parse.com SDK to allow for native push notifications.",
+	"min_platform_version": "v2.8.6",
+	"namespace": "parse",
+	"platform_version": "v2.8.6",
+	"version": "3.5.8"
```
Comment (rkusa) [inline module/manifest.json:14]: Why jumping from `3.4.0` to `3.5.8` and not to `3.5.0`?

**Review by rkusa — CHANGES_REQUESTED** [review]

File: module/manifest.json (hunk 1 above)
Comment (arblitroshani) [reply module/manifest.json:15]: File is auto-generated by Toolkit.
Will change 👍

File: module/manifest.json (hunk 3 above)
Comment (arblitroshani) [reply module/manifest.json:14]: At least to my knowledge and from what I searched online, there is no way to try a module without pushing it. 
And everytime you change the source code, you need to push a new version. 

I experimented with setting Trigger platform to `2.9.0`, without realizing it wasn't part of the task's ACs. We'll have to do it sometime soon anyways. 

I saw it is a normal workflow, based on previous version uploads (latest minors were `3.3.14` or `3.1.9`)

File: module/manifest.json (hunk 1 above)
Comment (arblitroshani) [reply module/manifest.json]: Force-pushed [this new commit](https://github.com/Staffbase/trigger.io-parse/pull/10/commits/b6f2e8e60beeacaef9692db72c1cbdc999a71b11) with the correct diff.

File: inspector/ios-inspector/dist/build_steps/sb_parse.json (hunk 2 above)
Comment (arblitroshani) [reply inspector/ios-inspector/dist/build_steps/sb_parse.json]: Fixed in [this commit](https://github.com/Staffbase/trigger.io-parse/pull/10/commits/8984c2625c92a793b02d28c78c4a65dff9c59cfa).

File: module/manifest.json (hunk 3 above)
Comment (rkusa) [reply module/manifest.json]: You can overwrite an existing version, but maybe only the latest one. This is at least how I did it in the past. But you are right it is then not possible anymore to go back.

**Review by rkusa — APPROVED** [review]

**Review by tomkretzschmar — APPROVED** [review]
Code review only (since it was tested twice already in the corresponding frontend PR)

**Pull request merged**